development/src/abilities/Swine Thug.js

```diff
@@ -100,57 +100,89 @@ G.abilities[37] =[
 			// For unupgraded ability, this is only 1 hex
 			// For upgraded, as long as the target is over a mud tile, keep pushing
 			// them back
-			var x = target.x;
-			var y = target.y;
-			var xLast = ability.creature.x;
-			var yLast = ability.creature.y;
-			var movementPoints = 0;
-			while(true) {
-				// Knock the target in the opposite direction of the attacker
-				var dx = x - xLast;
-				var dy = y - yLast;
+
+			// Calculate relative direction from creature to target
+			var dx = target.x - ability.creature.x;
+			var dy = target.y - ability.creature.y;
+			// Due to target size, this could be off; limit dx
+			if(dx > 1) dx = 1;
+			if(dx < -1) dx = -1;
+			var dir;
+			if (dy === 0) {
+				if (dx === 1) {
+					dir = 1;	// forward
+				} else {	// dx === -1
+					dir = 4;	// backward
+				}
+			} else {
 				// Hex grid corrections
-				if(dy !== 0) {
-					if(y%2 == 0) {
-						// Even row
-						dx++;
-					} else {
-						// Odd row
-						dx--;
+				if (ability.creature.y % 2 == 0 && dx < 1) {
+					dx++;
+				}
+				if (dx === 1) {
+					if (dy === -1) {
+						dir = 0;	// upright
+					} else {	// dy === 1
+						dir = 2;	// downright
+					}
+				} else {	// dx === 0
+					if (dy === 1) {
+						dir = 3;	// downleft
+					} else {	// dy === -1
+						dir = 5;	// upleft
 					}
 				}
-				// Due to target size, this could be off; limit dx
-				if(dx > 1) dx = 1;
-				if(dx < -1) dx = -1;
-				xLast = x;
-				yLast = y;
+			}
+			var hexes;
+			var flipped = false;
+			switch (dir) {
```

Comment:
Code duplication is usually a bad thing, perhaps the stuff should be split into a reusable function
https://github.com/FreezingMoon/AncientBeast/blob/master/development/src/abilities.js#L467